Source/ARTOSReachability.m

```diff
@@ -0,0 +1,87 @@
+//
+//  ARTReachability.m
+//  Ably
+//
+//  Created by Toni Cárdenas on 2/5/16.
+//  Copyright © 2016 Ably. All rights reserved.
+//
+
+#import <arpa/inet.h>
+#import <ifaddrs.h>
+#import <netdb.h>
+#import <sys/socket.h>
+#import <CoreFoundation/CoreFoundation.h>
+#import <SystemConfiguration/SystemConfiguration.h>
+#import <netinet/in.h>
+
+#import "ARTOSReachability.h"
+
+@implementation ARTOSReachability {
+    ARTLog *_logger;
+    NSString *_host;
+    void (^_callback)(BOOL);
+    SCNetworkReachabilityRef _reachabilityRef;
+    NSMutableDictionary *_instances;
+}
+
+- (instancetype)initWithLogger:(ARTLog *)logger {
+    self = [super self];
+    if (self) {
+        _logger = logger;
+        if (ARTOSReachability_instances == nil) {
+            _instances = [[NSMutableDictionary alloc] init];
+            ARTOSReachability_instances = _instances;
+        } else {
+            _instances = ARTOSReachability_instances;
+        }
+    }
+    return self;
+}
+
+__weak NSMutableDictionary *ARTOSReachability_instances;
+
+static void ARTOSReachability_Callback(SCNetworkReachabilityRef target, SCNetworkReachabilityFlags flags, void* info) {
+    [(ARTOSReachability *)ARTOSReachability_instances[[NSValue valueWithPointer:target]] internalCallback:flags & kSCNetworkReachabilityFlagsReachable];
+}
+
+- (void)listenForHost:(NSString *)host callback:(void (^)(BOOL))callback {
+    [self off];
+    _host = host;
+    _callback = callback;
+
+    _reachabilityRef = SCNetworkReachabilityCreateWithName(NULL, [host UTF8String]);
```

Comment:
It can be also done just for the connection. I thought why not make it more precise.